tests/commonTemplates_test.go

```diff
@@ -212,4 +218,60 @@ var _ = Describe("Common templates", func() {
 			table.Entry("[test_id:4770]golden image NS", &goldenImageNS),
 		)
 	})
+
+	Context("update old templates", func() {
+		BeforeEach(func() {
+			Expect(apiClient.Create(ctx, GetTestTemplate(oldVersion, strategy.GetNamespace()))).ToNot(HaveOccurred())
+			waitUntilDeployed()
+		})
+		AfterEach(func() {
+			Expect(apiClient.Delete(ctx, GetTestTemplate(oldVersion, strategy.GetNamespace()))).ToNot(HaveOccurred())
+		})
+		It("should remove labels from old templates", func() {
+			baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+			Expect(err).To(BeNil())
+			var versionRequirement *labels.Requirement
+			versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{commonTemplates.Version})
+			Expect(err).To(BeNil())
+			labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+			opts := client.ListOptions{
+				LabelSelector: labelsSelector,
+			}
+
+			var oldTemplates templatev1.TemplateList
+			err = apiClient.List(ctx, &oldTemplates, &opts)
+			Expect(err).To(BeNil())
+
+			for _, template := range oldTemplates.Items {
+				Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal(""), "os.template.kubevirt.io should be empty")
+				Expect(template.Labels["flavor.template.kubevirt.io/test"]).To(Equal(""), "flavor.template.kubevirt.io should be empty")
+				Expect(template.Labels["workload.template.kubevirt.io/server"]).To(Equal(""), "workload.template.kubevirt.io should be empty")
+				Expect(template.Labels["template.kubevirt.io/type"]).To(Equal("base"), "template.kubevirt.io/type should equal base")
+				Expect(template.Labels["template.kubevirt.io/version"]).To(Equal(oldVersion), "template.kubevirt.io/version should equal "+oldVersion)
+			}
+		})
+		It("should continue to have labels on latest templates", func() {
+			baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+			Expect(err).To(BeNil())
+			var versionRequirement *labels.Requirement
+			versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.Equals, []string{commonTemplates.Version})
+			Expect(err).To(BeNil())
+			labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+			opts := client.ListOptions{
+				LabelSelector: labelsSelector,
+			}
+
+			var oldTemplates templatev1.TemplateList
+			err = apiClient.List(ctx, &oldTemplates, &opts)
+			Expect(err).To(BeNil())
+
+			for _, template := range oldTemplates.Items {
+				Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal("true"), "os.template.kubevirt.io should not be empty")
```

Comment:
Same question as in the unit-tests, no template has `some-os` as a label

Reply:
Corrected now like unit-tests